compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -569,11 +599,42 @@ private BInvokableSymbol createSimpleVariable(BLangFunction function, BLangLambd
         BLangSimpleVariableDef variableDef = ASTBuilderUtil.createVariableDef(function.pos);
         variableDef.var = simpleVariable;
         variableDef.setBType(type);
+        if (isAnnotationClosure) {
+            annotationClosureReferences.add(variableDef);
+            return varSymbol;
+        }
         queue.add(variableDef);
+        return varSymbol;
+    }
 
+    public BVarSymbol createSimpleVariable(BInvokableSymbol invokableSymbol, boolean isAnnotationClosure) {
+        BType type = invokableSymbol.retType;
+        Location pos = invokableSymbol.pos;
+        Name name = invokableSymbol.name;
+        BVarSymbol varSymbol = new BVarSymbol(0, name, invokableSymbol.originalName, invokableSymbol.pkgID, type,
+                                              invokableSymbol.owner, pos, VIRTUAL);
+        BLangSimpleVariable simpleVariable = ASTBuilderUtil.createVariable(pos, name.value, type,
+                                                                           getInvocation(invokableSymbol), varSymbol);
+        BLangSimpleVariableDef variableDef = ASTBuilderUtil.createVariableDef(pos);
+        variableDef.var = simpleVariable;
+        variableDef.setBType(type);
+        if (isAnnotationClosure) {
+            annotationClosureReferences.add(variableDef);
+            return varSymbol;
+        }
+        queue.add(variableDef);
         return varSymbol;
```

Comment:
There seems to be some repeated logic with the previous method.Let's check and refactor separately.

Reply:
Updated(Change to private(https://github.com/ballerina-platform/ballerina-lang/pull/39349))